plugin/file/file.go

```diff
@@ -36,16 +36,35 @@ func (f File) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (i
 	state := request.Request{W: w, Req: r}
 
 	qname := state.Name()
-	// TODO(miek): match the qname better in the map
-	zone := plugin.Zones(f.Zones.Names).Matches(qname)
```

Comment:
Should this optimization be implemented in `Matches` instead?

Reply:
I will try to make another PR fixing tests failed, and implement search as a function. But It could not be done as Matches - as we use map of *Zone and the last one is defined in file plugin inside.